File: question12.py
# klavyeden girilen 20 adet sayidan cift olanlarin toplaminin tek olanlarin toplamina oranini bulan kod

tekTop=0
ciftTop=0

for i in range(0,20):
    sayi=int(input("Bir sayi giriniz: "))
    
    if sayi%2==0:
        ciftTop=ciftTop+sayi
    else:
        tekTop=tekTop+sayi

oran=ciftTop/tekTop

print("Cift sayilarin toplaminin tek sayilarin toplamina orani: " + str(oran))

File: question12.java
import java.util.Scanner;
public class question12 {
    public static void main(String[] args) {
        // klavyeden girilen 20 adet sayidan cift sayilarin toplaminin tek sayilarin toplamina oranini bulan kod

        Scanner scanner = new Scanner(System.in);

        int tekTop=0;
        int ciftTop=0;

        for(int i=0;i<20;i++){
            System.out.print("Sayi giriniz: ");
            int sayi=scanner.nextInt();
            
            if(sayi%2==0){
                ciftTop+=sayi;
            }
            else{
                tekTop+=sayi;
            }
        }

        double oran=ciftTop/tekTop;

        System.out.print("Cift sayilarin toplaminin tek sayilarin toplamina orani: " + oran);

        scanner.close();
    
    }
}
